src/Microsoft.SqlTools.ServiceLayer/BatchParser/BatchParserWrapper.cs

```diff
@@ -111,6 +117,73 @@ public sealed class BatchParserWrapper : IDisposable
             return batchDefinitionList;
         }
 
+        private static int GetMaxStartLine(IList<Tuple<int, int>> positions) 
+        {
+            int highest = 0;
+            foreach (var position in positions) 
+            {
+                if (position.Item1 > highest) 
+                {
+                    highest = position.Item1;
+                }
+            }
+            return highest;
+        }
+
+        /// <summary>
+        /// Gets offsets for all batches
+        /// </summary>
+        private static List<int> GetOffsets(string content, IList<Tuple<int, int>> positions) 
+        {
+
+            List<int> offsets = new List<int>();
+            int count = 0;
+            int offset = 0;
+            bool foundAllOffsets = false;
+            using (StringReader reader = new StringReader(content)) {
```

Comment:
there are still a couple '{' misplaced here and on line 151 for example.  Also '{' on 145 is misaligned.  Please double check formatting for this change.